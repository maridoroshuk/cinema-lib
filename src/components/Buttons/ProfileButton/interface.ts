export interface IProfileButton {
  onClick: () => void;
  children: string;
}
