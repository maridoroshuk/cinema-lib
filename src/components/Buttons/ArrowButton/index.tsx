import React from 'react';

import ArrowImage from '@/assets/icons/arrow.png';

import { Arrow, Wrapper } from './styles';
import { IArrowButton } from './interface';

export const ArrowButton = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Arrow src={ArrowImage} alt="arrow (hide)" />
    </Wrapper>
  );
};
