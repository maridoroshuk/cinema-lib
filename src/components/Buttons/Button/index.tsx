import React from 'react';

import { ButtonStyled } from '@/components/Buttons/Button/styles';

import { IButtonProps } from './interface';

export const Button = ({
  children,
  size,
  backgroundColor,
  type,
  value,
  onClick,
  dataCy,
}: IButtonProps) => {
  return (
    <ButtonStyled
      backgroundColor={backgroundColor}
      size={size}
      type={type || 'button'}
      value={value}
      onClick={onClick}
      data-cy={dataCy}
    >
      {children}
    </ButtonStyled>
  );
};
