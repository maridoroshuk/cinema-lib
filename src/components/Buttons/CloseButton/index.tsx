import React from 'react';
import { ICloseButtonProps } from './interface';
import { ButtonWrapper, StyledButton } from './styles';

export const CloseButton = ({ onClose }: ICloseButtonProps) => {
  return (
    <ButtonWrapper>
      <StyledButton onClick={onClose} />
    </ButtonWrapper>
  );
};
