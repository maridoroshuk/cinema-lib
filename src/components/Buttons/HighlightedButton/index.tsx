import React from 'react';
import { IUnderlineButton } from './interface';
import { ButtonStyled } from './styles';

export const HighlightedButton = ({
  children,
  handleButtonClick,
  value,
}: IUnderlineButton) => {
  return (
    <ButtonStyled onClick={handleButtonClick} value={value || ''}>
      {children}
    </ButtonStyled>
  );
};
