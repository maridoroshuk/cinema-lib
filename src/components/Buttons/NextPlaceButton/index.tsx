import React from 'react';

import NextPlaceButton from '@/assets/icons/next_place_button.png';

import { Next, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const NextPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Next src={NextPlaceButton} />
    </Wrapper>
  );
};
