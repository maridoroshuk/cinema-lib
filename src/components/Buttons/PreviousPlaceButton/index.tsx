import React from 'react';

import PreviousPlaceButton from '@/assets/icons/previous_place_button.png';

import { Previous, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const PreviousPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Previous src={PreviousPlaceButton} />
    </Wrapper>
  );
};
