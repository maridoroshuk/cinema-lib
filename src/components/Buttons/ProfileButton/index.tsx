import React from 'react';
import { IProfileButton } from './interface';
import { ButtonStyled } from './styles';

export const ProfileButton = ({ onClick }: IProfileButton) => {
  return <ButtonStyled onClick={onClick}>Profile</ButtonStyled>;
};
