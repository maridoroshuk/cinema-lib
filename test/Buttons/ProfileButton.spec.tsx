import React from 'react';
import { render } from '@testing-library/react';

import { ProfileButton } from '@/components/Buttons/ProfileButton';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('ProfileButton', () => {
  it('renders', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(<ProfileButton onClick={handleClick} />),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
