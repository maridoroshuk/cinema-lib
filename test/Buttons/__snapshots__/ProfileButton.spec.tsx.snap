// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileButton renders 1`] = `
.c0 {
  max-width: 280px;
  width: 100%;
  padding: 15px 30px;
  color: #FFFFFF;
  background-color: transparent;
  font-family: 'Poppins',sans-serif;
  line-height: 36px;
  font-weight: 300;
  font-size: 30px;
  border: none;
  border-radius: 10px;
  cursor: pointer;
  text-align: center;
  vertical-align: bottom;
}

.c0:after {
  width: 40px;
  height: 40px;
  content: '';
  display: inline-block;
  vertical-align: bottom;
  background-image: url();
  background-size: cover;
  margin-left: 20px;
}

<div
  class="c0"
>
  Profile
</div>
`;
